SECEdgar/tests/test_filings.py

```diff
@@ -63,11 +68,28 @@ def test_invalid_filing_type_types(self):
                 Filing(cik='0000320193', filing_type=t)
 
     def test_validate_cik(self):
-        with pytest.raises(CIKError):
+        # string remains unchecked until query to allow for possibility of
+        # using company name, ticker, or CIK as string
+        with pytest.raises(EDGARQueryError):
             Filing(cik='0notvalid0', filing_type=FilingType.FILING_10K)
-        with pytest.raises(CIKError):
+        with pytest.raises(EDGARQueryError):
             Filing(cik='012345678910', filing_type=FilingType.FILING_10K)
-        with pytest.raises(CIKError):
+        # float and int not accepted, raising TypeError from CIK Validator
+        with pytest.raises(TypeError):
             Filing(cik=1234567891011, filing_type=FilingType.FILING_10K)
-        with pytest.raises(ValueError):
+        with pytest.raises(TypeError):
             Filing(cik=123.0, filing_type=FilingType.FILING_10K)
+
+    def test_filing_save_multiple_ciks(self, multiple_valid_ciks, tmp_data_directory):
+        f = Filing(multiple_valid_ciks, FilingType.FILING_10Q, count=3)
+        f.save(tmp_data_directory)
+
+    def test_filing_save_single_cik(self, single_valid_cik, tmp_data_directory):
+        f = Filing(single_valid_cik, FilingType.FILING_10Q, count=3)
+        f.save(tmp_data_directory)
+
+    def test_filing_get_urls_returns_single_list_of_urls(self):
+        ciks = CIK(['aapl', 'msft', 'amzn'])
+        f = Filing(ciks, FilingType.FILING_10Q, count=3)
```

Comment:
So basically this test makes an actual network request? or we providing a mock object?

Reply:
Yes, it makes an actual network request. A lot of the tests do.